model/project_ref.go

```diff
@@ -326,6 +326,31 @@ func UntrackStaleProjectRefs(activeProjects []string) error {
 	return err
 }
 
+func (projectRef *ProjectRef) AsBsonM() bson.M {
```

Comment:
why do you need/want this?

I think you can just pass the struct itself to $set itself, but we should test it. (also if you have omitempty set on the fields, your update statement should also just work.